posixlib/src/main/scala/scala/scalanative/posix/stdio.scala

```diff
@@ -1,10 +1,91 @@
 package scala.scalanative
 package posix
 
-import scalanative.unsafe._
+import scalanative.unsafe, unsafe._
+import scalanative.posix.sys.types, types.{off_t, size_t}
 
 @extern object stdio extends stdio
 
 @extern trait stdio extends libc.stdio {
-  // no extensions
+  /* Open Group 2018 extensions to ISO/IEC C.
+   * Reference:
+   *   https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/stdio.h.html
+   *
+   * These definitions are annotated CX (ISO/IEC C extension)
+   * in the above specification.
+   */
+
+  type ssize_t = types.ssize_t
+
+  type va_list = unsafe.CVarArgList
+
+// Macros
+
+  /* Open Group POSIX defines this as a C macro.
+   * To provide the value in a portable manner, it is implemented here as
+   * an external method.  A slight but necessary deviation from the
+   * specification. The same idiom is used in an number of other posixlib
+   * files.
+   */
+  @name("scalanative_l_ctermid")
+  def L_ctermid: CUnsignedInt = extern
+
+// Methods
+
+  @blocking def ctermid(s: CString): CString = extern
+
+  @blocking def dprintf(fd: Int, format: CString, valist: va_list): Int = extern
+
+  @blocking def fdopen(fd: Int, mode: CString): Ptr[FILE] = extern
+  @blocking def fileno(stream: Ptr[FILE]): Int = extern
+  @blocking def flockfile(filehandle: Ptr[FILE]): Unit = extern
+
+  @blocking def fmemopen(
+      buf: Ptr[Byte],
+      size: size_t,
+      mode: CString
+  ): Ptr[FILE] = extern
+
+  @blocking def fseeko(stream: Ptr[FILE], offset: off_t, whence: Int): Int =
+    extern
+
+  @blocking def ftello(stream: Ptr[FILE]): off_t = extern
+  @blocking def ftrylockfile(filehandle: Ptr[FILE]): Int = extern
```

Comment:
The `ftrylockfile()` function is a nonblocking version of `flockfile()` so the annotation is not needed here. 
`@blocking` shall be used for methods that can block for indeterminate, possible long time, eg. getdelim which reads from stdin until, or any method that might use some underlying form of locking

Reply:
Good catch, thank you.   

I used to fail the "Which one is different from the others" drills in kindergarten also...